### PR TITLE
2 new mech equipments

### PR DESCRIPTION
**Pull request by Arokha**
Also a generic framework for atom/movables to cloak and uncloak.

The 2 equipment are an omni-directional shield generator (uses more power, only covers mech, when compared to the existing linear combat shield generator), and a cloaking device (uses gargantuan amounts of power, still visible on thermals).

I did this for some fighters I made downstream. You can have the fighters too if you want, but I didn't include them here.

dreamseeker_2020-03-23_20-58-20

### CODE REVIEW

**Comment by listerla** [comment]
now I just want to say I WANT THE FIGHTERS

**Comment by Mechoid** [comment]
I, too, am interested in the fighters.

**Comment by Arokha** [comment]
Ok, added fighters, unconstructable.

File: code/modules/mob/living/simple_mob/subtypes/animal/giant_spider/lurker.dm [hunk 1]
```diff
@@ -47,29 +47,12 @@
 
 	ai_holder_type = /datum/ai_holder/simple_mob/melee/hit_and_run
 
-	var/cloaked = FALSE
 	var/cloaked_alpha = 45			// Lower = Harder to see.
 	var/cloaked_bonus_damage = 30	// This is added on top of the normal melee damage.
 	var/cloaked_weaken_amount = 3	// How long to stun for.
 	var/cloak_cooldown = 10 SECONDS	// Amount of time needed to re-cloak after losing it.
 	var/last_uncloak = 0			// world.time
```
Comment (Novacat) [inline code/modules/mob/living/simple_mob/subtypes/animal/giant_spider/lurker.dm]:
```suggestion

/mob/living/simple_mob/animal/giant_spider/lurker/cloak()
	if(cloaked)
		return
	animate(src, alpha = cloaked_alpha, time = 1 SECOND)
	cloaked = TRUE


/mob/living/simple_mob/animal/giant_spider/lurker/uncloak()
	last_uncloak = world.time // This is assigned even if it isn't cloaked already, to 'reset' the timer if the spider is continously getting attacked.
	if(!cloaked)
		return
	animate(src, alpha = initial(alpha), time = 1 SECOND)
	cloaked = FALSE

```

**Pull request merged**